system/ee/ExpressionEngine/Boot/boot.php

```diff
@@ -61,7 +61,7 @@
  */
     try {
         if (file_exists(SYSPATH .'../.env.php')) {
-            $dotenv = ExpressionEngine\Dependency\Dotenv\Dotenv::createImmutable(SYSPATH .'../', '.env.php');
+            $dotenv = ExpressionEngine\Dependency\Dotenv\Dotenv::createUnsafeImmutable(SYSPATH .'../', '.env.php');
```

Comment:
Can someone explain why we are making this change?

Reply:
Good catch, it came from this commit (https://github.com/ExpressionEngine/ExpressionEngine/commit/2f4df0be4384e5b7809b8a3aa101216066ffee9e) but I don't know what issue Yuri was trying to fix.  It sounds like `createUnsafeImmutable` makes ENV vars available for native `getenv()` and `putenv()` calls.  So maybe reverting this and changing the line below from `getenv('EE_INSTALL_MODE')` to `$_ENV['EE_INSTALL_MODE'] ?? false` would be a good solution.

All the other `getenv()` calls I see in the core are for PHP env vars not user-defined variables

Reply:
Yeah, it just seemed like a big change for PHP 8.4 compatibility. Whatever option you think is better @bryannielsen